tools/apprunner-common/src/main/java/org/projectnessie/quarkus/runner/JavaVM.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.quarkus.runner;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Locale;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+import java.util.function.IntFunction;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Helper class to locate a JDK by Java major version and return the path to the {@code java}
+ * executable.
+ */
+public final class JavaVM {
```

Comment:
Given the way we are changing the plugin can we remove this code and just fail if we are <JDK11?

Reply:
The integration tests do fail when run with Java < 11.
I'd rather keep this one so we can test against newer Java versions, when we want to.
The code that actually checks for a different JDK/JRE is only run, if you're not running a Java 11 compatible JVM.

Reply:
That sounds to me like its dead code then? Or am i missing something? Realistically we will support only jdk11 until quarkus stops support. So the build will fail before we reach this point on jdk8 and on jdk11 we will not use this code.

please correct me if I don't get it.

Reply:
Not really dead. In case of Iceberg yes, but other projects that are strictly tied to a Java 8 build have to use this functionality.